lendingbot.py

```diff
@@ -26,562 +32,616 @@
+        print "!!! Please update to new config that includes Inactive Mode. !!!"  # Update alert.
+    apiKey = config.get("API", "apikey")
+    apiSecret = config.get("API", "secret")
+    minDailyRate = Decimal(config.get("BOT", "mindailyrate")) / 100
+    maxDailyRate = Decimal(config.get("BOT", "maxdailyrate")) / 100
+    spreadLend = int(config.get("BOT", "spreadlend"))
+    gapBottom = Decimal(config.get("BOT", "gapbottom"))
+    gapTop = Decimal(config.get("BOT", "gaptop"))
+    if config.has_option("BOT", "sixtyDayThreshold"):
+        sixtyDayThreshold = float(config.get("BOT", "sixtydaythreshold")) / 100
+        xDayThreshold = sixtyDayThreshold
+        xDays = "60"
+    else:
+        xDayThreshold = Decimal(config.get("BOT", "xdaythreshold")) / 100
+        xDays = str(config.get("BOT", "xdays"))
+    auto_renew = int(config.get("BOT", "autorenew"))
+    if config.has_option('BOT', 'outputCurrency'):
+        outputCurrency = config.get('BOT', 'outputCurrency')
+    if config.has_option('BOT', 'maxtolent'):
+        maxtolent = Decimal(config.get('BOT', 'maxtolent'))
+    if config.has_option('BOT', 'maxpercenttolent'):
+        maxpercenttolent = Decimal(config.get('BOT', 'maxpercenttolent')) / 100
+    if config.has_option('BOT', 'maxtolentrate'):
+        maxtolentrate = Decimal(config.get('BOT', 'maxtolentrate')) / 100
+    if config.has_option('BOT', 'minloansize'):
+        minLoanSize = Decimal(config.get("BOT", 'minloansize'))
+    if config.has_option('BOT', "transferableCurrencies"):
+        transferableCurrencies = (config.get("BOT", "transferableCurrencies")).split(",")
+    else:
+        transferableCurrencies = []
+
+    if config.has_option("BOT", "coinconfig"):
+        try:
+            # parsed
+            coinconfig = (json.loads(config.get("BOT", "coinconfig")))
+            for cur in coinconfig:
+                cur = cur.split(':')
+                coincfg[cur[0]] = dict(minrate=(Decimal(cur[1])) / 100, maxactive=Decimal(cur[2]),
+                                       maxtolent=Decimal(cur[3]), maxpercenttolent=(Decimal(cur[4])) / 100,
+                                       maxtolentrate=(Decimal(cur[5])) / 100)
+        except Exception as e:
+            print "Coinconfig parsed incorrectly, please refer to the documentation. Error: " + e
+            pass
+
+sleepTime = sleepTimeActive  # Start with active mode
+# sanity checks
+if sleepTime:
+    if sleepTime < 1 or sleepTime > 3600 or sleepTimeInactive < 1 or sleepTimeInactive > 3600:
+        print "sleeptime values must be 1-3600"
+        exit(1)
+if minDailyRate:
+    if minDailyRate < 0.00003 or minDailyRate > 0.05:  # 0.003% daily is 1% yearly
+        print "mindaily rate is set too low or too high, must be 0.003-5%"
+        exit(1)
+if maxDailyRate:
+    if maxDailyRate < 0.00003 or maxDailyRate > 0.05:
+        print "maxdaily rate is set too low or too high, must be 0.003-5%"
+        exit(1)
+if spreadLend:
```

Comment:
I'll be honest, I don't like the sanity checks. I feel like they need to be done during parsing itself or right before the respective variable is used (I recommend the former). Right now, PyCharm sees a logical possibility that the sanity checks could run without first defining the variables and throws several errors. I would like to get that changed asap.